dags/common/rest_api_client.py

```diff
@@ -1,13 +1,15 @@
-class RestApi:
+class RestApiClient:
```

Comment:
Now that I see it, this class is called `RestApiClient` but the usage that you have inside it is the one of a `Request`. That's why you are setting parameters and headers, you are treating this class are a `Request`. I think this is fine but you must chose a path between both. I think for our use case, It's better to go with the Request. So we have : 
- common/request
  - Here we have an object that defines that gets the base path, port, request parameters and headers
  - We can then run a request. The error handling should be done in this class, and the result should be returned
- aps/APSApiClient
  - This client uses the request class when triggering requests. The error handling is in the request object, so the class can focus on only getting the data when it exists and doing what it must with it